s2core/src/main/scala/com/kakao/s2graph/core/Graph.scala

```diff
@@ -319,11 +319,20 @@ object Graph {
 
       val successCallback = (kvs: util.ArrayList[KeyValue]) => {
         val edgeWithScores = Edge.toEdges(kvs, queryParam, prevScore, isInnerCall = false, parentEdges)
-        QueryResult(q, stepIdx, queryParam, edgeWithScores)
+        val sample = queryParam.sample
+        val sampledEdgesWithScores = if (sample >= 0 ) {
+          // sample only from non-degree edges
+          val (degreeEdges, edges) = edgeWithScores.partition { case (edge, _) => edge.propsWithTs.contains(LabelMeta.degreeSeq) }
```

Comment:
I think this is O(N) operations. if we are traversing large number of edges, then this can be performance bottleneck. 
degree edge is only exist on head of edgegs only if queryParam.offset == 0, otherwise there is no degree edge. 

what do you think?